fs/fs_zenfs.cc

```diff
@@ -805,8 +805,49 @@ IOStatus ZenFS::RenameFileNoLock(const std::string& source_path,
       files_.insert(std::make_pair(source_path, source_file));
     }
   } else {
+    std::string source_seperator = (source_path.back() != '/') ? "/" : "";
+    std::string dest_seperator = (dest_path.back() != '/') ? "/" : "";
+    std::vector<std::string> children;
+    s = GetChildrenNoLock(source_path, options, &children, dbg);
+    if (!s.ok()) {
+      return s;
+    }
+
+    for (const auto& child : children) {
+      s = RenameFileNoLock(source_path + source_seperator + child,
+                           dest_path + dest_seperator + child, options, dbg);
+      if (!s.ok()) {
+        /* roll back renameing */
+        for (const auto& rollback_child : children) {
+          s = RenameFileNoLock(dest_path + dest_seperator + rollback_child,
+                               source_path + source_seperator + rollback_child,
+                               options, dbg);
+          if (!s.ok()) {
+            return IOStatus::Corruption(
+                "ZenFS RenameFileNoLock: Unable to roll back unsuccessful "
+                "renameing.");
+          }
+        }
+        return s;
+      }
+    }
+
```

Comment:
Since RocksDB moved to c++17 I think it makes sense to use `std::filesystem::path::append` to concatenate components of the paths.

Reply:
@percona-ysorokin clarified that percona-server release-8.0.28-19 enables c++17 without the need to switch to RocksDB 7.0.x.